adal/adal-rules.pro

```diff
@@ -0,0 +1,65 @@
+# Add project specific ProGuard rules here.
+# By default, the flags in this file are appended to flags specified
+# in C:\Program Files\Android\android-sdk/tools/proguard/proguard-android.txt
+# You can edit the include path and order by changing the proguardFiles
+# directive in build.gradle.
+#
+# For more details, see
+#   http://developer.android.com/guide/developing/tools/proguard.html
+
+# Add any project specific keep options here:
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
+
+-dontskipnonpubliclibraryclasses
+
+##---------------Begin: proguard configuration for Common  --------
+-keep,includedescriptorclasses class com.microsoft.identity.common.exception.** { *; }
+-keep,includedescriptorclasses class com.microsoft.identity.common.internal.broker.** { *; }
```

Comment:
I'm thinking if we could just do 

-keep,includedescriptorclasses class com.microsoft.identity.common.**

This way, we don't have to maintain this file (and the one in MSAL) much when we introduce new stuffs in common.

Reply:
The idea here is that we don't use most of common in ADAL, so we should drop as much of it as we can.